### PR TITLE
test: Add unit tests for KubeFlow sub-groupers and PodJob grouper

### PR DESCRIPTION
**Pull request by kube-gopher**
## Summary
This PR adds unit tests for five grouper plugins that previously had 0% 
coverage, bringing them in line with the rest of the KubeFlow grouper family

## Changes
- `pkg/podgrouper/podgrouper/plugins/kubeflow/jax`
- `pkg/podgrouper/podgrouper/plugins/kubeflow/notebook`
- `pkg/podgrouper/podgrouper/plugins/kubeflow/tensorflow`
- `pkg/podgrouper/podgrouper/plugins/kubeflow/xgboost`
- `pkg/podgrouper/podgrouper/plugins/podjob`

## Test coverage
| Package | Before | After |
|---------|--------|-------|
| kubeflow/jax | 0% | ~100% |
| kubeflow/notebook | 0% | ~85.7% |
| kubeflow/tensorflow | 0% | ~100% |
| kubeflow/xgboost | 0% | ~100% |
| podjob | 0% | ~100% |

## How to verify
```bash
go test ./pkg/podgrouper/podgrouper/plugins/kubeflow/... -v -cover
go test ./pkg/podgrouper/podgrouper/plugins/podjob/... -v -cover
```

## Notes
- Tests follow the existing patterns established in `kubeflow/pytorch` and 
  `kubeflow/mpi`
- Uses Ginkgo v2 + Gomega, consistent with the rest of the codebase
- No production code was modified

## Checklist
- [x] Self-reviewed
- [x] Added/updated tests (if needed)
- [ ] Updated documentation (if needed)

## Summary by CodeRabbit

* **Tests**
  * Expanded test coverage for pod grouper plugins across Kubeflow and PodJob implementations, including validation of pod grouping metadata, replica configuration handling, priority class selection, queue resolution, topology extraction, and label/annotation inheritance from job specifications.

### CODE REVIEW

**Review by enoodle — APPROVED** [review]

**Comment by enoodle** [comment]
@kube-gopher There is no need to update the PR - the merge queue will take care of this.

**Comment by kube-gopher** [comment]
@enoodle Thanks for the reminder. I’ll let the merge queue handle it.

**Review by davidLif — APPROVED** [review]
